build.rs

```diff
@@ -19,34 +19,32 @@ fn print_link_search_path() {}
 
 #[cfg(feature = "with-syntex")]
 mod macros {
-    extern crate syntex;
     extern crate pnet_macros;
-
+    extern crate syntex;
+    extern crate glob;
+    
     use std::env;
     use std::path::Path;
-
-    const FILES: &'static [&'static str] = &["ethernet.rs",
-                                             "gre.rs",
-                                             "ipv4.rs",
-                                             "ipv6.rs",
-                                             "icmp.rs",
-                                             "udp.rs",
-                                             "tcp.rs",
-                                             "arp.rs",
-                                             "vlan.rs"];
-
+    
     pub fn expand() {
-        let out_dir = env::var_os("OUT_DIR").unwrap();
-
-        for file in FILES {
-            let src_file = format!("src/packet/{}.in", file);
-            let src = Path::new(&src_file);
-            let dst = Path::new(&out_dir).join(file);
-
-            let mut registry = syntex::Registry::new();
-            pnet_macros::register(&mut registry);
-
-            registry.expand("", &src, &dst).unwrap();
+        // globbing for files to pre-process:
+        let pattern = "./**/*.rs.in";
+        for entry in glob::glob( pattern ).expect("Failed to read glob pattern") {
+            match entry {
+                Ok(path) => {
+                    println!(" CMP-- {}", path.display() );
+                    //src: Path::new() = /full/path/file.rs.in
+                    let src     = Path::new( path.to_str().expect("Invalid src Specified.") );
+                    //src -> dst: Path::new() = OUT_DIR/file.rs
+                    let out_dir = env::var_os( "OUT_DIR" ).expect("Invalid OUT_DIR.");
+                    let file    = Path::new( path.file_stem().expect("Invalid file_stem.") );
+                    let dst     = Path::new( &out_dir ).join(file);
+                    let mut registry = syntex::Registry::new();
+                    pnet_macros::register(&mut registry);
+                    registry.expand("", &src, &dst).unwrap();
+                },
+                Err(_) => {},
```

Comment:
Since you're ignoring the error case, this `match` would probably be better as an `if let` instead.